pkg/apis/unversioned/codec/json_codec.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2015 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+//package unversioned_codec
+
+package codec
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+
+	"k8s.io/kubernetes/pkg/runtime"
+)
+
+type jsonCodec struct{}
```

Comment:
@liggitt Those methods are not called for these objects--they are not even implemented for this codec.

@caesarxuchao I think it is a good idea to make the unversioned objects more "real".

@liggitt / @deads2k Will you be satisfied with TODO items here?